0006/README.md

```diff
@@ -0,0 +1,319 @@
+
+Git branch is the way to manage versioned specifications.
+
+*What is changing*
+
+-   `release/<version|version-draft>` is the branch pattern for a prior or
+    future `version`, where the `version` must contain both major and minor.
+-   In addition to the shared images in the `main` branch, if an image is
+    required for a specific version, the `/images` directory in the version
+    branch may contain the image and let the versioned files reference it .
+-   At the top level directory, a branch has all the versioned file from
+    `/docs/spec/<version>/*`, `/docs/provenance/<version>.md`,
+    `/docs/verification_summary/<version>.md`, and
+    `/docs/github-actions-workflow/<version>.md`. All specification files in the
+    same branch share the same version.
+
+## Automate Building and Deploying The Website
+
+A shell script should assemble the website, instead of using any logic in the
+Github Workflow script.
+
+### Checkout repo, load all submodules, and build once
+
+The GitHub Workflow clones the `main` branch once, pulls the latest files for
+all the submodules, builds the website once, and deploys the website to Github
+Pages.
+
+*Disadvantages*
+
+-   The usage of Git submodule is non-standard, and it leaves some versioned
+    directories in the `main` branch.
+
+*What is changing and code snapshot*
+
+The following is a snapshot of the scripts for building and deploying the
+website.
+
+-   To serve the website locally,
+    [/docs/README.md](https://github.com/slsa-framework/slsa/blob/main/docs/README.md)
+    is replaced with running `build.sh` or similar logic.
+-   The Github Workflow uses the same shell script to prepare the build
+    environment.
+    
+```
+#!/bin/bash
+#
+# build.sh lives in .github/scripts. This script is preliminary and for
+# illustrative purposes. It may live in a repo other than slsa.
+# 
+# - By default, serves the SLSA website on your local machine.
+# - Using an argument, prepares the code environment without serving.
+
+serve_website=${s:-"true"} # Gets value from the command argument.
+
+if [[ -d "slsa" ]]; then
+  rm -rf slsa
+fi
+
+sudo apt install ruby ruby-dev bundler
+git clone --recurse-submodules https://github.com/chtiangg/slsa.git
+cd slsa/docs
+git submodule update --remote
+bundle config set --local path 'vendor/bundle'
+bundle install
+
+if [[ "${serve_website}" == "true" ]]; then
+  echo "Start to serve the website..."
+  bundle exec jekyll serve --livereload --incremental
+else
+  echo "Finish setting up code and installing dependencies. Exit."
+fi
+
+--------------------------------------------------------------------------
+
+# deploy_website.yml lives in .github/workflows.
+# This script is preliminary and for illustrative purposes. It may live in
+# a repo other than slsa.
+
+# ... Prepares the environment for Github Workflow
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - name: Run the setup script
```

Comment:
For the rest of this, can we just use the https://github.com/actions/jekyll-build-pages? I think the only special thing we need to do to get our setup working is the `submodules: recursive` bit, right?